Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM busybox:stable-glibc as builder
+RUN echo "nobody:x:65534:65534:Nobody:/:" > /etc/nobody
+
+FROM scratch
+COPY --from=builder /etc/nobody /etc/passwd
+USER nobody
+COPY kubectl-slice /kubectl-slice
+
+ENTRYPOINT ["/kubectl-slice"]
```

Comment:
While I see what you did here 😆 I think this would be better if:

* It's a `debian` image, you can use `bookworm` from here https://hub.docker.com/_/debian
* Adds a user, say, `slice`
* Adds that user to the `sudoers` so they can do `sudo`
* Installs `sudo` (and after installing it, clears the `apt` cache)
* Uses `USER slice` to downgrade the current shell to `slice` levels, not to `root`
* Still supports using `sudo` after being downgraded

A quick idea I put together is:

```suggestion
# Use specific versions for reproducibility
ARG KUBECTL_VERSION=1.31.1
ARG YQ_VERSION=v4.44.3

# Stage 1: Download binaries
FROM debian:12-slim AS downloader

ARG KUBECTL_VERSION
ARG YQ_VERSION

RUN apt-get update && apt-get install -y --no-install-recommends \
  curl \
  ca-certificates \
  && rm -rf /var/lib/apt/lists/*

# Download kubectl
RUN curl -sSL -o /kubectl "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl" \
  && chmod +x /kubectl

# Download yq
RUN curl -sSL -o /yq "https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64" \
  && chmod +x /yq

# Base imagen for kubectl-slice
FROM debian:12-slim

ARG DEBIAN_FRONTEND=noninteractive

# Install necessary packages and create user
RUN apt-get update && apt-get install -y --no-install-recommends \
  sudo \
  && useradd -m -s /bin/bash slice \
  && echo 'slice ALL=(ALL) NOPASSWD:ALL' > /etc/sudoers.d/slice \
  && chmod 0440 /etc/sudoers.d/slice \
  && rm -rf /var/lib/apt/lists/*

# Copy binaries from the downloader stage
COPY --from=downloader /kubectl /usr/local/bin/kubectl
COPY --from=downloader /yq /usr/local/bin/yq
COPY --from=downloader /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt

# Copy kubectl-slice from local filesystem
COPY kubectl-slice /usr/local/bin/kubectl-slice

USER slice
WORKDIR /home/slice
```

Reply:
Hey Patrick,thank you for the review :) ,i forgot about kubectl,had a doubt why a debian not an alpine ?

Reply:
No problem! Was about to type a review here but I think it's easier if I leave you my notes in context. I'm marking this one as resolved for the time being.